api/src/events.py

```diff
@@ -0,0 +1,625 @@
+    event['end_date'] = datetime.fromisoformat(request.json['end_date'])
+    event['num_registrations'] = 0
+    event['event_participants'] = []
+    event['description'] = request.json['description'] if 'description' in request.json else None
+    event['is_virtual'] = request.json['is_virtual']
+    event['location'] = request.json['location'] if 'location' in request.json else None
+    event['zoom_link'] = request.json['zoom_link'] if 'zoom_link' in request.json else None
+
+    db.events.insert_one(event)
+
+    return jsonify({"msg": "event added"}), 200
+
+
+    
+@events_api.route('/', methods=['DELETE'])
+@jwt_required
+@check_admin
+def delete_event():
+
+    """Deletes an event.
+
+    :reqjson event_name: event name
+    :reqjson display_name: name of event displayed to general public
+
+    Example input:
+
+    .. sourcecode:: json
+
+    {
+        "event_name": "HopHacks_Fall_2022",
+    }
+
+    :status 200: Event deleted
+    :status 400: Error with request or operation
+    
+    """
+
+    if (request.json is None):
+        return Response('Data not in json format', status=400)
+
+    if not (all(field in request.json for field in ['event_name'])):
+        return Response('Invalid request', status=400)
+    event = {}
+    event['event_name'] = request.json['event_name']
+    if len(list(db.events.find(event))) == 0:
+        return Response('Event Does Not Exist', status=400)
+
+    db.events.delete_one(event)
+    return jsonify({"msg": "event deleted"}), 200
+
+    
+@events_api.route('/', methods=['PUT'])
+@jwt_required
+@check_admin
+def update_event():
+
+    """Updates an event.
+
+    :reqjson event_name: event name
+    :reqjson new_event_name: new event name that event_name is being changed to (optional)
+    :reqjson display_name: new name of event displayed to general public (optional)
+    :reqjson start_date: new start date of event (optional)
+    :reqjson end_date: new end date of event (optional)
+    :reqjson description: new description of event (optional)
+    :reqjson is_virtual: new boolean string true/false saying whether event is virtual or in-person (optional)
+    :reqjson location: new location of event (optional)
+    :reqjson zoom_link: new zoom link of event (optional)
+
+    Example input:
+
+    .. sourcecode:: json
+
+    {
+        "event_name": "HopHacks_Fall_2022",
+        "new_event_name": "Hophacks_Fall_2023"
+    }
+
+    :status 200: Event updated
+    :status 400: Error with request or update operation
+
+    """
+
+    if (request.json is None):
+        return Response('Data not in json format', status=400)
+
+    if not (all(field in request.json for field in ['event_name'])):
+        return Response('Invalid request', status=400)
+
+    event_current = {}
+    event_current['event_name'] = request.json['event_name']
+
+    if len(list(db.events.find(event_current))) == 0:
+        return Response('Event Does Not Exist', status=400)
+
+    for i in request.json:
```

Comment:
nitpick: use variable name 'field' or other name, instead of i, i is usually for number